internals/controllers/rolebinding/rolebinding_helpers.go

```diff
@@ -0,0 +1,129 @@
+package controllers
+
+import (
+	"github.com/dana-team/hns/internals/utils"
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+// IsValidRoleBinding checks if a RoleBinding object is valid.
+func IsValidRoleBinding(roleBinding client.Object) bool {
+	if !isRoleBinding(roleBinding) {
+		return false
+	}
+
+	if len(roleBinding.(*rbacv1.RoleBinding).Subjects) == 0 {
+		return false
+	}
+
+	rbKind := roleBinding.(*rbacv1.RoleBinding).Subjects[0].Kind
+	if rbKind == "Group" {
+		rbName := roleBinding.(*rbacv1.RoleBinding).Subjects[0].Name
+		if strings.HasPrefix(rbName, "system") {
+			return false
+		}
+	}
+
+	if rbKind == "ServiceAccount" {
+		if roleBinding.(*rbacv1.RoleBinding).Subjects[0].Namespace == "" {
+			return false
+		}
+	}
+
+	rbName := roleBinding.(*rbacv1.RoleBinding).Subjects[0].Name
+	if rbName == "default" || rbName == "builder" || rbName == "deployer" {
```

Comment:
you need to store this strings as an enum or cost that's the deffinition to magic string and your name is not harry potter

Reply:
changed it